File: codes/cpp/01/01-04/01-04-02.cpp
// https://www.luogu.com.cn/problem/B2138
#include <bits/stdc++.h>
using namespace std;

const int MAXN = 5005;
int maxp[MAXN];  

void init() {
    memset(maxp, 0, sizeof(maxp));
    for (int i = 2; i < MAXN; ++i) {
        if (maxp[i] == 0) { 
            maxp[i] = i;
            for (int j = i * 2; j < MAXN; j += i) {
                maxp[j] = i; 
            }
        }
    }
}

int main() {
    init();
    int m, n;
    cin >> m >> n;
    for (int i = m; i <= n; ++i) {
        if (i > m) cout << ",";
        cout << maxp[i];
    }
    return 0;
}

File: codes/python/01/01-04/01-04-02.py
# https://www.luogu.com.cn/problem/B2138
MAXN = 5005
maxp = [0] * MAXN
for i in range(2, MAXN):
    if maxp[i] == 0:  
        maxp[i] = i
        for j in range(i * 2, MAXN, i):
            maxp[j] = i
m, n = map(int, input().split())
res = [str(maxp[x]) for x in range(m, n + 1)]
print(",".join(res))